src/software/simulation/simulator.h

```diff
@@ -189,8 +189,8 @@ class Simulator
      */
     static void updateSimulatorRobots(
         const std::vector<std::weak_ptr<PhysicsRobot>>& physics_robots,
-        std::map<std::shared_ptr<SimulatorRobot>, std::shared_ptr<FirmwareWorld_t>>&
-            simulator_robots,
+        std::map<std::shared_ptr<PhysicsSimulatorRobot>,
```

Comment:
does this have to use `PhysicsSimulatorRobot`?

Reply:
Yes, since in `Simulator::setBallState`, `clearBallInDribblerArea` is called on the robots in the map and this isn't an interface needed by `SslSimulatorRobots`